addon/components/rdf-input-fields/numerical-input.js

```diff
@@ -1,18 +1,51 @@
 import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 import { guidFor } from '@ember/object/internals';
 import SimpleInputFieldComponent from '@lblod/ember-submission-form-fields/components/rdf-input-fields/simple-value-input-field';
 import { XSD } from '@lblod/submission-form-helpers';
-import { literal } from 'rdflib';
+import { Literal, literal } from 'rdflib';
+import { hasValidFieldOptions } from '@lblod/ember-submission-form-fields/utils/has-valid-field-options';
+import { FORM_OPTION } from '@lblod/ember-submission-form-fields/utils/namespaces';
 
 export default class RdfInputFieldsNumericalInputComponent extends SimpleInputFieldComponent {
   inputId = 'input-' + guidFor(this);
 
+  @tracked allowClear = false;
+
+  constructor() {
+    super(...arguments);
+
+    this.loadOptions();
+  }
+
+  loadOptions() {
+    let allowClear = this.args.formStore.any(
+      this.args.field.uri,
+      FORM_OPTION('allowClear'),
+      undefined,
+      this.args.graphs.formGraph
+    );
+    if (!allowClear) {
+      if (!hasValidFieldOptions(this.args.field, ['allowClear'])) {
+        return;
```

Comment:
nitpick:
Do we want to set this.allowClear  equal to false here? I always find it difficult when a value is defined on init.
```js
@tracked allowClear:
loadOptions() { 
  if (!hasValidFieldOptions(this.args.field, ['allowClear'])) {
        this.allowClear = false;
  }
}

Reply:
fair point